### PR TITLE
[EoC]Add some meta functions

### PR DESCRIPTION
**Pull request by lispcoc**
#### Summary
None

#### Purpose of change
I will add functions that I thought should exist but is not implemented, which I personally thought while developing the MOD.

#### Describe the solution
function
- `message` - `u_message`, `npc_message` displays messages only if talker is avatar. This function always displays messages.

condition
- `u_is_avatar`, `npc_is_avatar` - Check if u or npc is avatar.
- `u_is_npc`, `npc_is_npc` - Check if u or npc is NPC.
- `u_is_character`, `npc_is_character` - Check if u or npc is character.
- `u_is_monster`, `npc_is_monster` - Check if u or npc is monster.
- `u_is_item`, `npc_is_item` - Check if u or npc is item.
- `u_is_furniture`, `npc_is_furniture` - Check if u or npc is furniture.
- `player_see_u`, `player_see_npc` - Check if player (game view) can see u or npc.
- `u_can_see_location`, `npc_can_see_location` - Check if u or npc can see the location.

#### Testing
Add CI test and examples.

### CODE REVIEW

**Comment by GuardianDll** [comment]
As a use case, this `{ "or": [ "npc_male", "npc_female" ] }` can be replaced with one of the functions
https://github.com/CleverRaven/Cataclysm-DDA/blob/7ae6d0f2166e8c2bef19dcb8aef8063ce6fbcfd5/data/json/monsters/zed_lieutenant.json#L43-L45
@RenechCDDA which fit it better - targeting only avatar, or both avatar and npcs? maybe we need a separate function `u_is_avatar_or_npc`?

**Comment by RenechCDDA** [comment]
The `u_is_character` function would handle that, actually.

image

All NPCs (and the player avatar) are characters, and all characters are also creatures. But characters are NOT monsters.

File: doc/EFFECT_ON_CONDITION.md [hunk 1]
```diff
@@ -2805,12 +2816,12 @@ adds [ your strength stat ] amount of faction trust
...
 
-#### `u_message`, `npc_message`
-Display a text message in the log
+#### `u_message`, `npc_message`, `message`
+Display a text message in the log. `u_message` and `npc_message` display a message only if you or NPC is avatar. `message` always displays a message.
 
 | Syntax | Optionality | Value  | Info |
 | --- | --- | --- | --- | 
-| "u_message" / "npc_message" | **mandatory** | string or [variable object](##variable-object) | default true |
+| "u_message" / "npc_message" / "message" | **mandatory** | string or [variable object](##variable-object) | default true |
```
Comment (GuardianDll) [inline doc/EFFECT_ON_CONDITION.md:2824]:
```suggestion
| "u_message" / "npc_message" / "message" | **mandatory** | string or [variable object](##variable-object) | message, that would be printed; If `snippet` is true, id of a snippet that would be printed |
```
How lucky for you to find field, that i forgot to document XD

File: doc/EFFECT_ON_CONDITION.md [hunk 2]
```diff
@@ -283,9 +283,20 @@ return true if alpha talker is female
 "condition": "npc_female",
 ```
 
-return true if beta talker is male or female; return false, if talker is not capable to have a gender (if monster, for example, can be used if you want to target only player or NPC)
+### `u_is_avatar`, `u_is_npc`, `u_is_character`, `u_is_monster`, `u_is_item`, `u_is_furniture`, `npc_is_avatar`, `npc_is_npc`, `npc_is_character`, `npc_is_monster`, `npc_is_item`, `npc_is_furniture`
+- type: simple string
+- return true if alpha or beta talker is avatar / NPC / character / monster / item /furniture
+
```
Comment (GuardianDll) [inline doc/EFFECT_ON_CONDITION.md:289]:
```suggestion
- `avatar` is you, player, that control specific NPC (yes, your character is still NPC, you just can control it, as you can control another NPC using faction succession)
- `npc` is any NPC, except Avatar
- `character` is both NPC or Avatar

```

**Comment by GuardianDll** [comment]
if we enter the `player_see_u` territory, can you consider to resolve #67370, as make the field accept tripoint coordinate?

**Comment by lispcoc** [comment]
> if we enter the `player_see_u` territory, can you consider to resolve #67370, as make the field accept tripoint coordinate, delivered from `_location_variable` effect?

`player_see_u` was created to implement a function equivalent to `add_msg_if_player_sees` in C++. Since the subject of this function is always the player, I don't think adding the function of specifying coordinates is sufficient to solve #67370.
I would like to implement another function that mainly uses `u` and `n` instead, but I think that replacing `u_can_see` and `npc_can_see` as they are will have a big impact.
I think it would be better to implement a separate function with a name like `u_can_see_location`, `npc_can_see_location`

**Pull request merged**